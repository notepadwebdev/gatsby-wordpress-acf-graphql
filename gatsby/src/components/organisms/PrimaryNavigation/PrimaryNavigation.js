import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'gatsby';
import './PrimaryNavigation.scss';

const PrimaryNavigation = ({ menuItems }) => {
  return (
    <nav className="primary-navigation" role="navigation">
      <ul>
        {menuItems.nodes.map((node) => {
          return (
            <li key={node.id}>
              <Link to={node.path} activeClassName="active" partiallyActive={true}>
                {node.label}
              </Link>
            </li>
          );
        })}
      </ul>
    </nav>
  );
};

export default PrimaryNavigation;

PrimaryNavigation.propTypes = {
  menuItems: PropTypes.shape({
    nodes: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.string.isRequired,
        label: PropTypes.string.isRequired,
        path: PropTypes.string.isRequired
      })
    )
  }).isRequired
};
