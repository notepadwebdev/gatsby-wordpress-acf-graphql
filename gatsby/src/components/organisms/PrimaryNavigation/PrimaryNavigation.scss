.primary-navigation {
    ul {
        margin: 0;
        padding: 0;
        list-style: none;
        display: flex;
        align-items: center;
        justify-content: center;
    }

    li {
        padding: 1em;
    }

    a.active {
        text-decoration: none;
        color: inherit;
        pointer-events: none;
    }
}
