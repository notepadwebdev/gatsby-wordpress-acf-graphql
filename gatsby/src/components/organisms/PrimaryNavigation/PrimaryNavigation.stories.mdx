import { Meta, Story, Canvas, ArgsTable } from '@storybook/addon-docs/blocks';
import PrimaryNavigation from './PrimaryNavigation';

<Meta
  title="Organisms/PrimaryNavigation"
  component={PrimaryNavigation}
/>

export const Template = (args) => <PrimaryNavigation {...args} />;

export const Default = Template.bind({});
Default.args = {
  menuItems: {
    nodes: [
      {
        id: `bmF2X21lbnVfaXRlbTo4`,
        label: `Menu Item 1`,
        path: `/menu-item-1/`
      },
      {
        id: `bmF2X21lbnVfaXRlbToxNg==`,
        label: `Menu Item 2`,
        path: `/menu-item-2/`
      }
    ]
  }
}

# PrimaryNavigation

The main navigation for the site.

<ArgsTable of={PrimaryNavigation} />

<Canvas>
  <Story name="Default" args={Default.args}>
    {Template.bind({})}
  </Story>
</Canvas>
