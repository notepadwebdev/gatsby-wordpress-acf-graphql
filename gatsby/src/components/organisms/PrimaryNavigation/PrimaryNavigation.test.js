import React from 'react';
import renderer from 'react-test-renderer';
import PrimaryNavigation from './PrimaryNavigation';

describe(`Primary Navigation component`, () => {
  it(`renders correctly`, () => {
    const component = renderer.create(
      <PrimaryNavigation menuItems={{
        nodes: [
          {
            id: `bmF2X21lbnVfaXRlbTo4`,
            label: `Menu Item 1`,
            path: `/menu-item-1/`
          },
          {
            id: `bmF2X21lbnVfaXRlbToxNg==`,
            label: `Menu Item 2`,
            path: `/menu-item-2/`
          }
        ]
      }} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
