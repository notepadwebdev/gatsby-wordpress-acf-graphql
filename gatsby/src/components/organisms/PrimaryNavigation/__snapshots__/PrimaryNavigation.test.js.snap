// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Primary Navigation component renders correctly 1`] = `
<nav
  className="primary-navigation"
  role="navigation"
>
  <ul>
    <li>
      <a
        href="/menu-item-1/"
      >
        Menu Item 1
      </a>
    </li>
    <li>
      <a
        href="/menu-item-2/"
      >
        Menu Item 2
      </a>
    </li>
  </ul>
</nav>
`;
